From eec40b77bad18b1b239444107e419e6cf0a90252 Mon Sep 17 00:00:00 2001
From: jerryshao <sshao@hortonworks.com>
Date: Wed, 6 Jul 2016 14:49:21 +0100
Subject: [PATCH 392/517] [MINOR][CORE][1.6-BACKPORT] Fix display wrong free
 memory size in the log

## What changes were proposed in this pull request?

Free memory size displayed in the log is wrong (used memory), fix to make it correct. Backported to 1.6.

## How was this patch tested?

N/A

Author: jerryshao <sshao@hortonworks.com>

Closes #14043 from jerryshao/memory-log-fix-1.6-backport.

(cherry picked from commit 2588776ad3d91e39300d61c8a12dc0803c28e866)
---
 .../org/apache/spark/storage/MemoryStore.scala     |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/core/src/main/scala/org/apache/spark/storage/MemoryStore.scala b/core/src/main/scala/org/apache/spark/storage/MemoryStore.scala
index 17aae6e..aed0da9 100644
--- a/core/src/main/scala/org/apache/spark/storage/MemoryStore.scala
+++ b/core/src/main/scala/org/apache/spark/storage/MemoryStore.scala
@@ -393,7 +393,8 @@ private[spark] class MemoryStore(blockManager: BlockManager, memoryManager: Memo
         }
         val valuesOrBytes = if (deserialized) "values" else "bytes"
         logInfo("Block %s stored as %s in memory (estimated size %s, free %s)".format(
-          blockId, valuesOrBytes, Utils.bytesToString(size), Utils.bytesToString(blocksMemoryUsed)))
+          blockId, valuesOrBytes, Utils.bytesToString(size),
+          Utils.bytesToString(maxMemory - blocksMemoryUsed)))
       } else {
         // Tell the block manager that we couldn't put it in memory so that it can drop it to
         // disk if the block allows disk storage.
-- 
1.7.9.5

